Panic when executing `kubectl directpv drives access-tier unset` command

**Describe the bug**
a runtime panic occurs when `kubectl directpv drives access-tier unset` command is called

**To Reproduce**
Just by calling `kubectl directpv drives access-tier unset --all`

**Expected behavior**
Should unset any access tiers set on the drives

**Screenshots and logs**

```log
ubuntu@directpv-1:~$ k directpv drives access-tier unset --all
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x145619f]

goroutine 1 [running]:
main.processFilteredDrives.func2(0xc000bef180)
	github.com/minio/directpv/cmd/kubectl-directpv/utils.go:159 +0xff
github.com/minio/directpv/pkg/client.ProcessDrives.func2({0x1a1b128?, 0xc000bef180?})
	github.com/minio/directpv/pkg/client/client.go:334 +0x36
github.com/minio/directpv/pkg/client.processObjects({0x1a2c3a8?, 0xc0000b69c0}, 0xc000739d10?, 0xc000c8fa58, 0xc000c8fa68, 0xc0002c7730, {0x1a176a0, 0xc000a11668}, 0x0)
	github.com/minio/directpv/pkg/client/client.go:201 +0x375
github.com/minio/directpv/pkg/client.ProcessDrives({0x1a2c3a8?, 0xc0000b69c0}, 0xc00018c840, 0xc00070fb40, 0x18c2c98, 0x18c2500, {0x1a176a0, 0xc000a11668}, 0x0?)
	github.com/minio/directpv/pkg/client/client.go:330 +0x1c5
main.processFilteredDrives({0x1a2c3a8, 0xc0000b69c0}, {0x0?, 0x0?, 0xc0005c29f0?}, 0x0, 0x7?, 0x17?, {0x17eb5f8, 0xf})
	github.com/minio/directpv/cmd/kubectl-directpv/utils.go:152 +0x286
main.unsetAccessTier({0x1a2c3a8?, 0xc0000b70c0?})
	github.com/minio/directpv/cmd/kubectl-directpv/drives_accesstier_unset.go:92 +0xa5
main.glob..func3(0x2565360, {0x17dcdee?, 0x1?, 0x1?})
	github.com/minio/directpv/cmd/kubectl-directpv/drives_accesstier_unset.go:75 +0x3ac
github.com/spf13/cobra.(*Command).execute(0x2565360, {0xc0002c7020, 0x1, 0x1})
	github.com/spf13/cobra@v1.1.3/command.go:852 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0x25646e0)
	github.com/spf13/cobra@v1.1.3/command.go:960 +0x39c
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.1.3/command.go:897
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra@v1.1.3/command.go:890
main.Execute(...)
	github.com/minio/directpv/cmd/kubectl-directpv/cmd.go:135
main.main()
	github.com/minio/directpv/cmd/kubectl-directpv/kubectl-directpv.go:45 +0x16f
```


**Deployment information (please complete the following information):**
 - DirectPV version: v3.0.3
 - Kubernetes Version: any
 - OS info: any
- Kernel version: any

**Additional context**
Add any other context about the problem here.

--- a/cmd/kubectl-directpv/utils.go
+++ b/cmd/kubectl-directpv/utils.go
@@ -54,6 +54,7 @@ var (
 	globRegexp                  = regexp.MustCompile(`(^|[^\\])[\*\?\[]`)
 	errMixedSelectorUsage       = errors.New("either glob or ellipsis pattern is supported")
 	errMixedStatusSelectorUsage = fmt.Errorf("either glob or [%s] is supported", strings.Join(directcsi.SupportedStatusSelectorValues(), ", "))
+	errEmptyApplyProcessFn      = errors.New("apply and process functions can't be nil")
 )
 
 // Command defines the directpv drives sub-command
@@ -122,6 +123,11 @@ func processFilteredDrives(
 ) error {
 	var resultCh <-chan client.ListDriveResult
 	var err error
+
+	if applyFunc == nil || processFunc == nil {
+		return errEmptyApplyProcessFn
+	}
+
 	if len(idArgs) > 0 {
 		resultCh = getDrivesByIds(ctx, idArgs)
 	} else {
@@ -155,10 +161,10 @@ func processFilteredDrives(
 		ctx,
 		resultCh,
 		func(drive *directcsi.DirectCSIDrive) bool {
-			return drive.MatchGlob(
-				nodeGlobs,
-				driveGlobs,
-				statusGlobs) && matchFunc(drive)
+			if match := drive.MatchGlob(nodeGlobs, driveGlobs, statusGlobs); match {
+				return matchFunc == nil || matchFunc(drive)
+			}
+			return false
 		},
 		applyFunc,
 		processFunc,
@@ -357,6 +363,11 @@ func processFilteredVolumes(
 ) error {
 	var resultCh <-chan client.ListVolumeResult
 	var err error
+
+	if applyFunc == nil || processFunc == nil {
+		return errEmptyApplyProcessFn
+	}
+
 	if len(idArgs) > 0 {
 		resultCh = getVolumesByIds(ctx, idArgs)
 	} else {
@@ -389,11 +400,13 @@ func processFilteredVolumes(
 		ctx,
 		resultCh,
 		func(volume *directcsi.DirectCSIVolume) bool {
-			return volume.MatchNodeDrives(nodeGlobs, driveGlobs) &&
+			if match := volume.MatchNodeDrives(nodeGlobs, driveGlobs) &&
 				volume.MatchPodName(podNameGlobs) &&
 				volume.MatchPodNamespace(podNsGlobs) &&
-				volume.MatchStatus(volumeStatusList) &&
-				matchFunc(volume)
+				volume.MatchStatus(volumeStatusList); match {
+				return matchFunc == nil || matchFunc(volume)
+			}
+			return false
 		},
 		applyFunc,
 		processFunc,

